lib/pacemaker/pcmk_sched_group.c

```diff
@@ -359,42 +361,58 @@ pcmk__group_internal_constraints(pcmk_resource_t *rsc)
  * \param[in]     primary        Primary resource in colocation
  * \param[in]     colocation     Colocation constraint to apply
  */
-static void
+static int
 colocate_group_with(pcmk_resource_t *dependent, const pcmk_resource_t *primary,
                     const pcmk__colocation_t *colocation)
 {
-    pcmk_resource_t *member = NULL;
+    int priority_delta = 0;
 
     if (dependent->priv->children == NULL) {
-        return;
+        return 0;
     }
 
     pcmk__rsc_trace(primary, "Processing %s (group %s with %s) for dependent",
                     colocation->id, dependent->id, primary->id);
 
     if (pe__group_flag_is_set(dependent, pcmk__group_colocated)) {
         // Colocate first member (internal colocations will handle the rest)
-        member = (pcmk_resource_t *) dependent->priv->children->data;
-        member->priv->cmds->apply_coloc_score(member, primary, colocation,
-                                              true);
-        return;
+        pcmk_resource_t *member = dependent->priv->children->data;
+        priority_delta = member->priv->cmds->apply_coloc_score(member, primary,
+                                                               colocation,
+                                                               true);
+
+        if (priority_delta != 0) {
+            dependent->priv->priority =
+                pcmk__add_scores(priority_delta, dependent->priv->priority);
```

Comment:
This could use a comment explaining why it's needed

Keep the commits separate but swap them (first changing method return value, second with complete fix)